crates/next-core/src/app_structure.rs

```diff
@@ -940,6 +955,38 @@ fn directory_tree_to_loader_tree_internal(
         }
     }
 
+    // make sure we don't have a match for other slots if there's an intercepting route match
+    if tree.is_intercepting() {
+        let mut keys_to_replace = Vec::new();
+
+        for (key, parallel_tree) in &tree.parallel_routes {
+            if !parallel_tree.is_intercepting() {
+                keys_to_replace.push(key.clone());
+            }
+        }
+
+        for key in keys_to_replace {
+            tree.parallel_routes.insert(
+                key,
+                LoaderTree {
+                    page: app_page.clone(),
+                    segment: "__DEFAULT__".into(),
+                    parallel_routes: IndexMap::new(),
+                    components:
+                        // default fallback component
+                        Components {
+                            default: Some(
+                                get_next_package(app_dir)
+                                    .join("dist/client/components/parallel-route-default.js".into()),
+                            ),
```

Comment:
Maybe move this into a helper function, since you're constructing this default fallback component in multiple places now?

Reply:
Unfortunately, none of those are exactly the same